SharedPackages/BrowserServicesKit/Sources/Common/DDGError.swift

```diff
@@ -0,0 +1,277 @@
+    /// - Free of sensitive information (passwords, tokens, etc.)
+    /// - Consistent in format and terminology
+    ///
+    /// Example:
+    /// ```swift
+    /// var description: String {
+    ///     switch self {
+    ///     case .connectionTimeout: return "Network request timed out after 30 seconds"
+    ///     case .invalidCredentials: return "Authentication failed - invalid credentials"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// - Note: For user-facing error messages, implement `DDGErrorUIPresentable`.
+    var description: String { get }
+}
+
+// MARK: - Error Chain Traversal
+
+/// Extensions providing error chain traversal and debugging capabilities.
+public extension DDGError {
+
+    /// Returns an array containing this error and all underlying errors in the chain.
+    ///
+    /// This property traverses the complete error chain, starting from the current error
+    /// and following the `underlyingError` links until reaching an error with no underlying cause.
+    ///
+    /// ## Chain Traversal Rules
+    /// - The first element is always the current error (self)
+    /// - Subsequent elements are underlying errors in order
+    /// - Traversal stops when an error has no `underlyingError`
+    /// - Traversal stops when encountering a non-DDGError (for safety)
+    ///
+    /// ## Performance
+    /// This is computed each time it's accessed. For repeated access, consider caching the result.
+    ///
+    /// - Returns: An array of errors representing the complete error chain.
+    /// - Complexity: O(n) where n is the depth of the error chain.
+    var errorsChain: [Error] {
+        var errors: [Error] = []
+        var currentError: Error? = self
+
+        while let error = currentError {
+            errors.append(error)
+
+            if let ddgError = error as? any DDGError {
+                currentError = ddgError.underlyingError
+            } else {
+                break
+            }
+        }
+
+        return errors
+    }
+
+    /// Returns a formatted debug string representation of the complete error chain.
+    ///
+    /// This property creates a multi-line string showing all errors in the chain,
+    /// with each error on its own line prefixed with a dash. It's designed for
+    /// debugging and logging purposes.
+    ///
+    /// ## Output Format
+    /// ```
+    /// - Primary error description
+    /// - Underlying error description
+    /// - Root cause description
+    /// ```
+    ///
+    /// ## Description Selection
+    /// - For DDGError instances: Uses the `description` property
+    /// - For other Error types: Uses Swift's default string representation
+    ///
+    /// - Important: This is intended for debugging only. Do not display this to users.
+    /// - Returns: A formatted multi-line string representing the error chain.
+    var errorsChainDebugDescription: String {
+        let errorsDescriptions = errorsChain.map({ error in
```

Comment:
This could be wrapped in an `#if DEBUG` to ensure it only gets used in debug mode